File: Daily/PY/Leetcode678-有效的括号字符串.py
class Solution:
    def checkValidString(self, s: str) -> bool:
        cnt1,cnt2 = 0,0
        for i in range(len(s)):
            if s[i] == '(': 
                cnt1 += 1
                cnt2 += 1
            elif s[i] == '*':
                cnt1 -= 1
                cnt2 += 1
            else:
                cnt1 -= 1
                cnt2 -= 1
            cnt1 = max(0,cnt1)
            if cnt1 > cnt2: return False    
        return cnt1 == 0

File: Leetcode/sbgo/Leetcode678-有效的括号字符串.go
package sbgo

func checkValidString(s string) bool {
	cnt1, cnt2 := 0, 0
	for i := 0; i < len(s); i++ {
		if s[i] == '(' {
			cnt1++
			cnt2++
		} else if s[i] == '*' {
			cnt1--
			cnt2++
		} else {
			cnt1--
			cnt2--
		}
		cnt1 = max(cnt1, 0)
		if cnt1 > cnt2 {
			return false
		}
	}
	return cnt1 == 0
}
